components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -258,15 +273,59 @@ func (r *Reconciler) syncSubscriptionStatus(ctx context.Context, sub *eventingv1
 	if !reflect.DeepEqual(sub.Status, desiredSubscription.Status) {
 		err := r.Client.Status().Update(ctx, desiredSubscription, &client.UpdateOptions{})
 		if err != nil {
+			events.EventWarn(r.recorder, desiredSubscription, events.ReasonUpdateFailed, "Update Subscription status failed %s", desiredSubscription.Name)
 			return errors.Wrapf(err, "update subscription status failed")
 		}
+		events.EventNormal(r.recorder, desiredSubscription, events.ReasonUpdate, "Update Subscription status succeeded %s", desiredSubscription.Name)
 	}
 	return nil
 }
 
-func (r *Reconciler) assertSinkValidity(sink string) error {
-	_, err := url.ParseRequestURI(sink)
-	return err
+func DefaultSinkValidator(r *Reconciler, ctx context.Context, subscription *eventingv1alpha1.Subscription) error {
```

Comment:
Could you please add a unit-test for this function?

Reply:
Added